### DISCUSSION TITLE
[@types/styled-components] Type 'typeof X' is not assignable to type 'ComponentClass<any, any>'.

### DISCUSSION THREAD

**Discussion by ericlifs** [DefinitelyTyped/DefinitelyTyped#57173] [Issues with a @types package]
Hi everyone, I'm having some issues with styled-components and react-native-linear-gradient.

I'm currently having my monorepo created using https://github.com/mmazzarolo/react-native-universal-monorepo. I have no issues with any other components, except when I trying to style a react-native-linear-gradient component by doing something like:

```
export const BackgroundGradient = styled(LinearGradient)`
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
`;
```

On my browser I'm getting the following error:
<img width="475" alt="image" src="https://user-images.githubusercontent.com/7376892/141860337-478b80c1-3876-47f8-b82b-15918b49ed3f.png">

I've already searched for similar issues and found [this one](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/31538) but I have already checked that I only have one unique `@types/react` version installed (17.0.34) so that solution doesn't apply to my issue.

Packages versions:

- @types/react: "17.0.34"
- @types/styled-components-react-native: "5.1.3"
- @types/styled-components: "5.1.15"

Thank you very much!

[a comment by an automated account is not shown]

**Reply by lifeiscontent** [reply]
Do you mind posting the code for the linear gradient as well? It would help debug in case there is an issue with types merging

**Reply by ericlifs** [reply]
I didn't do anything special regarding the linear gradient, it's just having that BackgroundGradient defined in a file:

```
export const BackgroundGradient = styled(LinearGradient)`
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
`;
```

And in your react component file just import & use it: 

```
<BackgroundGradient
  colors={["#13141C", "f3058a"]}
  locations={[0.1, 1]}
  start={{ x: 0.5, y: 1 }}
  end={{ x: 0.5, y: 0 }}
/>
```

**Reply by lifeiscontent** [reply]
@ericlifs I was specifically talking about the code for `LinearGradient`
